chatkit/src/main/java/com/pusher/chatkit/ChatManager.kt

```diff
@@ -103,23 +115,31 @@ class ChatManager @JvmOverloads constructor(
 
     private fun ChatKitEvent.handleEvent() {
         when (this) {
-            is CurrentUserReceived -> this@ChatManager.currentUser = currentUser
+            is CurrentUserReceived -> currentUserPromiseContext.report(currentUser.asSuccess())
             is ErrorOccurred -> logger.error(error.reason, error.asSystemError())
             is RoomDeleted -> roomStore -= roomId
             is RoomUpdated -> roomStore += room
-            is CurrentUserRemovedFromRoom -> currentUser?.id?.let { id -> roomStore[roomId]?.removeUser(id) }
+            is CurrentUserRemovedFromRoom -> currentUser.onReady { result ->
+                result.map { currentUser -> roomStore[roomId]?.removeUser(currentUser.id) }
+            }
+            is CurrentUserAddedToRoom -> currentUser.onReady { result ->
+                result.map { currentUser -> room.addUser(currentUser.id) }
+            }
```

Comment:
Are there still some events being missed here? https://github.com/pusher/chatkit-client-js/blob/4da9cd6871ced4b9b6d830082a89d5fbfabdf21f/src/user-subscription.js#L33-L60

User being added / removed from a room and typing indicators seem to be missing

Reply:
Yeah, there are a few live events that aren't there yet. Adding cards for those